libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -1513,12 +1513,18 @@ public void onEditorThemeUpdated(Bundle editorTheme) {
         getGutenbergContainerFragment().mediaFileSaveProgress(localId, progress);
     }
 
-    @Override public void onMediaSaveFailed(String localId) {
+    @Override public void onMediaSaveFailed(String localId, float progress) {
         getGutenbergContainerFragment().mediaFileSaveFailed(localId);
         mFailedMediaIds.add(localId);
         mUploadingMediaProgressMax.remove(localId);
     }
 
+    @Override public void onMediaUploadPaused(String localId, float progress) {
+        getGutenbergContainerFragment().mediaFileSaveFailed(localId);
```

Comment:
If I interpret this correctly, I believe this should use the new pause method and pass progress through  to methods called on its way to the JS app. Passing progress may become unnecessary, depending on your response to [my feedback](https://github.com/WordPress/gutenberg/pull/57476#discussion_r1439793155) on the sibling PR. 

```suggestion
        getGutenbergContainerFragment().mediaFileSavePaused(localId, progress);
```

At this point, it is unclear to me where we would invoke this new `onMediaUploadPaused` method. From quickly reviewing this code in Android Studio, it appears `onMediaSaveFailed` is never utilized anywhere. There must be other locations that manage and communicate failed media uploads to the JS app.

Reply:
>  From quickly reviewing this code in Android Studio, it appears onMediaSaveFailed is never utilized anywhere. There must be other locations that manage and communicate failed media uploads to the JS app.

This is what I was trying to figure out as well. Currently, the paused and failed handlers are very similar, in that they pass `localId` and `progress`. The latter progress `param` was added to `onMediaSaveFailed` this PR to explore if displaying progress could handled via that method instead of the paused state. I would like to match the iOS implementation more closely, and will steer this code toward that, but it seems the existing platform implementations for handling failed media uploads diverge from each other quite a bit.

Reply:
> At this point, it is unclear to me where we would invoke this new onMediaUploadPaused method.

@dcalhoun  I explored this further, and now it is also unclear to me where these listeners are invoked. It may occur in the FluxC repository. While the `enum` solution in iOS is straightforward, it seems like an Android implementation may be much more complex and may require the addition of several paused event handlers to essentially communicate the same boolean value that we already have access to: is the network connected or not?

With a nod to the original spirit of the task being focused on updating the media upload progress visuals rather than the behavior, at the moment a more Javascript-focused solution may be less complex than introducing an entire new upload state.

Reply:
@derekblank it would appear that `onMediaSaveFailed` was added for the Story block in https://github.com/wordpress-mobile/WordPress-Android/pull/12982 (in addition to all of the "save" methods/events) and does not directly relate to uploads for other media-related blocks (e.g., Image blocks). Most media-related blocks rely upon the `onMediaUpload*` methods/events. 

Curiously, the one location invoking `onMediaSaveFailed` was [disabled](https://github.com/wordpress-mobile/WordPress-Android/commit/8e415babbf281f77d11b057b8e01d3145ea78aa3) and remains in a [commented state](https://github.com/wordpress-mobile/WordPress-Android/blob/234db297728a506ee8e9c318d42c9cd9e7e941b3/WordPress/src/main/java/org/wordpress/android/ui/posts/editor/StoriesEventListener.kt#L212). 

I did not research why new "save" events were required for the Story block, but stopped my investigation when I realized they are unrelated.